### PR TITLE
improve connect-boost feature

### PR DESCRIPTION
**Pull request by arvidn**
to make new torrents quickly connect peers

### CODE REVIEW

File: include/libtorrent/torrent.hpp [hunk 1]
```diff
@@ -1610,7 +1603,13 @@ namespace libtorrent
 		// the number of bytes of padding files
 		boost::uint32_t m_padding:24;
 
-		// TODO: gap of 8 bits available here
+		// this is set to the connect boost quota for this torrent.
+		// After having received this many priority peer connection attempts, it
+		// fallse back onto the steady state peer connection logic, driven by the
+		// session tick.  The first tracker response, when this is non-zero will
+		// attempt to connect to a bunch of peers immediately and set this to
+		// false. We only do this once to get the torrent kick-started
```
Comment (ssiloti) [inline include/libtorrent/torrent.hpp]: This comment needs (more) updating, the variable is not longer immediately set to false.

**Comment by Lash78** [comment]
@arvidn 
Can't compile RC_1_1 branch now

`src\torrent.cpp(298): error C2614: 'libtorrent::torrent': illegal member initialization: 'm_connect_boost_counter' is not a base or member
src\torrent.cpp(3761): error C2065: 'm_connect_boost_counter': undeclared identifier
src\torrent.cpp(3766): error C2065: 'm_connect_boost_counter': undeclared identifier
src\torrent.cpp(3766): error C2672: 'std::min': no matching overloaded function found
src\torrent.cpp(3767): error C2780: '_Ty std::min(std::initializer_list<_Elem>)': expects 1 arguments - 2 provided
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.14.26428\include\algorithm(5370): note: see declaration of 'std::min'
src\torrent.cpp(3767): error C2780: 'const _Ty &std::min(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)': expects 3 arguments - 2 provided
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.14.26428\include\algorithm(5344): note: see declaration of 'std::min'
src\torrent.cpp(3778): error C2065: 'm_connect_boost_counter': undeclared identifier
src\torrent.cpp(9954): error C2065: 'm_connect_boost_counter': undeclared identifier`

**Comment by arvidn** [comment]
hm. it built on msvc on appveyor:

https://ci.appveyor.com/project/arvidn/libtorrent/build/1.0.9714

**Comment by arvidn** [comment]
are you sure your `torrent.hpp` file is updated?

**Comment by Lash78** [comment]
yes, torrent.hpp updated too. i'm trying build for qbittorrent VS2017, boost1.67. Everything was fine before this commit.
here is the command:
`b2 -q --without-python --toolset=msvc-14.1 address-model=64 variant=release link=static runtime-link=static debug-symbols=on encryption=on logging=off resolve-countries=off dht=on character-set=unicode boost-link=static -sBOOST_ROOT="G:\qBittorrent\boost_1_67_0" include="G:\QBITTORRENT\install_msvc64\base\include" library-path="G:\QBITTORRENT\install_msvc64\base\lib" --prefix="G:\QBITTORRENT\install_msvc64\base" cxxflags="-O1 -Gy -Gw -GL" define=BOOST_ASIO_DISABLE_CONNECTEX linkflags="/NOLOGO /DYNAMICBASE /NXCOMPAT /LTCG /OPT:REF /OPT:ICF /MANIFEST:EMBED /INCREMENTAL:NO" --hash -j %NUMBER_OF_PROCESSORS%`

**Comment by ssiloti** [comment]
RC_1_1 builds fine for me with msvc 14.1. You surely have an old version of `torrent.hpp` being pulled in somehow.

**Comment by Lash78** [comment]
Ok, sry, there was same old .hpp files in the include path, that was the problem, ok now..

**Comment by arvidn** [comment]
perhaps the connect_boost default setting should be lower. I can't think of an obvious mechanism for it to distinguish a single new torrent being added and starting up with lots of torrents. the latter case probably does better without any boost, as connections are spread evenly

**Pull request merged**